game/src/mydungeon/ECS.java

```diff
@@ -50,12 +53,14 @@ protected void setup() {
         new KeyboardSystem();
         new AISystem();
         new CollisionSystem();
+        new HealthSystem();
     }
 
     @Override
     protected void frame() {
         camera.setFocusPoint(heroPositionComponent.getPosition());
-
+        entities.remove(entitiesToRemove);
```

Comment:
```suggestion
        entities.removeAll(entitiesToRemove);
```

Würde remove nicht das Set an sich versuchen aus entities zu entfernen, aber nicht die Elemente von entitiesToRemove. Glaube, du wolltest hier ein removeAll nutzen, welches alle Elemente von entitiesToRemove aus entities entfernt. Kann mich dabei aber auch irren.

Reply:
haste recht, ist gefixt